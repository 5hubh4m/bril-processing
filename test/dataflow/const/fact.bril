@main {
  result: int = const 1;
  i: int = const 8;
.header:
  zero: int = const 0;
  cond: bool = gt i zero;
  br cond .body .end;
.body:
  result: int = mul result i;
  one: int = const 1;
  i: int = sub i one;
  jmp .header;
.end:
  print result;
}
