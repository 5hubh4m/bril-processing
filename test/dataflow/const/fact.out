main:
  l1f6f3:
    in: âˆ…
    out: result: 1, i: 8
  header:
    in: zero: 0, one: 1
    out: zero: 0, one: 1
  body:
    in: zero: 0, one: 1
    out: zero: 0, one: 1
  end:
    in: zero: 0, one: 1
    out: zero: 0, one: 1
